pkg/systemd/services/service.jsx

```diff
@@ -21,50 +21,132 @@
+            { interface: s_bus.I_MANAGER, member: "Reloading" },
+            async (_path, _iface, _signal, [reloading]) => {
+                debug("Service detail", this.props.unitId, "Reloading", reloading);
+                this.setState({ reloading });
+                if (!reloading)
+                    await this.updateProperties();
+            }));
+
+        this.subscriptions.push(dbus.subscribe(
+            { path: this.path, interface: s_bus.I_PROPS, member: "PropertiesChanged" }, async () => {
+                debug("Service detail", this.props.unitId, "PropertiesChanged");
+                await this.updateProperties();
+            }));
+
+        /* We need this *only* to pick up failed Conditions after attempting to start a service, as the unit immediately gets
+         * unloaded again and we don't get a PropertiesChanged signal. */
+        this.subscriptions.push(dbus.subscribe(
+            { interface: s_bus.I_MANAGER, member: "JobRemoved" }, async (_p, _i, _s, [_job_id, _job_path, unit, result]) => {
+                if (unit === this.props.unitId && result === "done") {
+                    debug("Service detail", this.props.unitId, "JobRemoved");
+                    await this.updateProperties();
+                }
+            }));
+
+        debug("Service detail", this.props.unitId, "componentDidMount, loading initial properties");
+        await this.updateProperties();
     }
 
-    componentDidMount() {
-        if (this.state.shouldFetchProps)
-            this.props.getUnitByPath(this.props.unit.path).finally(() => this.setState({ shouldFetchProps: false }));
+    componentWillUnmount() {
+        this.subscriptions.forEach(s => s.remove());
+        this.subscriptions = [];
+        debug("Service detail", this.props.unitId, "componentWillUnmount");
+    }
+
+    async updateProperties() {
+        const dbus = this.props.dbusClient;
+        const unit_id = this.props.unitId;
+
+        try {
+            const [unit_props] = await dbus.call(this.path, s_bus.I_PROPS, "GetAll", [s_bus.I_UNIT]);
+            // unwrap variants
+            for (const key in unit_props)
+                unit_props[key] = unit_props[key].v;
+
+            if (unit_id.endsWith(".timer")) {
+                const [timer_props] = await dbus.call(this.path, s_bus.I_PROPS, "GetAll", [s_bus.I_TIMER]);
+                this.props.addTimerProperties(timer_props, unit_props);
+            }
+
+            if (unit_id.endsWith(".socket")) {
+                const [socket_props] = await dbus.call(this.path, s_bus.I_PROPS, "GetAll", [s_bus.I_SOCKET]);
+                unit_props.Listen = socket_props.Listen.v;
+            }
+
+            unit_props.path = this.path;
+
+            debug("Service detail", unit_id, "updated properties:", JSON.stringify(unit_props));
+            this.setState({ unit_id, unit_props, error: null });
+        } catch (ex) {
+            this.setState({ error: ex.toString() });
+        }
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230308-084716-374aaf58-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)